src/Core/System/SalesChannel/Entity/SalesChannelRepository.php

```diff
@@ -91,7 +91,7 @@ public function search(Criteria $criteria, SalesChannelContext $salesChannelCont
 
         $ids = $this->doSearch($criteria, $salesChannelContext);
 
-        $readCriteria = $criteria->cloneForRead($ids->getIds());
+        $readCriteria = $criteria->cloneForRead(empty($ids->getIds()) ? null : $ids->getIds());
```

Comment:
No read should happen if there are no ids.

Reply:
No search happening, I hope I implemented a good alternative